tests/tests-common.h

```diff
@@ -530,3 +530,20 @@ static void checkBuild(Slice s, ValueType t, ValueLength byteSize) {
       break;
   }
 }
+
+#if __cplusplus >= 201703L
+
+template<typename T, typename U>
+bool haveSameOwnership(std::shared_ptr<T> const& left, std::shared_ptr<U> const& right) {
+  using cmp_ptr_type = std::shared_ptr<void const>;
```

Comment:
Do we need this type?
https://wandbox.org/permlink/4fAkr3oTgzPSMQyw
I am not against the thread_local comparator.

Reply:
Thanks, that makes it simpler.